gematria/datasets/find_accessed_addrs_exegesis.cc

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/datasets/find_accessed_addrs_exegesis.h"
+
+#include "BenchmarkRunner.h"
+#include "LlvmState.h"
+#include "Target.h"
+#include "TargetSelect.h"
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86RegisterInfo.h"
+#include "gematria/llvm/disassembler.h"
+
+using namespace llvm;
+using namespace llvm::exegesis;
+
+namespace gematria {
+
+ExegesisAnnotator::ExegesisAnnotator(
+    LlvmArchitectureSupport &ArchSup, LLVMState &ExegesisState,
```

Comment:
Good point. I've switched to only using `LLVMState` and storing a couple additional necessary components that would normally be held by `LlvmArchitectureSupport` so that we can keep everything consolidated to one (at least within the class).